plugins/postgresql_alt.py

```diff
@@ -44,6 +57,59 @@ def normalize_string(s):
 
         return '\n'.join([f'{key} = {normalize_string(value)}' for key, value in config])
 
+    @classmethod
+    def get_should_collect_logs(cls, conn: object) -> Tuple[LoggingInfo, Optional[Exception]]:
+        logging_info = LoggingInfo(False, '', '')
+        try:
+            with conn.cursor() as cur:
+                cur.execute('''
+                    SELECT name, setting
+                    FROM pg_settings
+                    WHERE name IN (
+                        'log_destination',
+                      , 'logging_collector'
+                      , 'log_directory'
+                      , 'data_directory'
+                    )''')
+                logging_config = cur.fetchall()
+                logging_config = {key:value for key, value in logging_config}
+
+                log_destinations = logging_config['log_destination'].split(',')
+                logging_collector = logging_config['logging_collector']
+
+                logging_info.log_dir = logging_config['log_directory']
+                logging_info.data_dir = logging_config['data_directory']
+        except psycopg2.Error as err:
+            return (logging_info, err)
+
+        except KeyError as err:
+            return (logging_info, err)
+
+        if any(item in ['stderr', 'csvlog'] for item in log_destinations):
+            if logging_collector == 'on':
+                logging_info.collect_logs = True
+
+        return (logging_info, None)
+
+    @classmethod
+    def docker_get_data_dir_host(cls, container_id: str, pg_data_dir: str) -> Tuple[str, Optional[Exception]]:
+        inspect_cmd  = "docker inspect "
+        inspect_cmd += "-f '{{range .Mounts }}{{println .Destination .Source}}{{ end }}' "
```

Comment:
Nitpick: why are there sometimes spaces inside the curlys, sometimes not? Is this part of the syntax?

Reply:
I call that laziness ;-)